From 3bd0505973c4dc7877cf6b8b3a0ba475c89f9bb9 Mon Sep 17 00:00:00 2001
From: Dan Pasanen <dan.pasanen@gmail.com>
Date: Tue, 22 Aug 2017 23:22:38 -0500
Subject: [PATCH] btm_inq: fix build with BTA_HOST_INTERLEAVE_SEARCH

* Need to explicitly cast the int to btm_inq_state enum value
  after incrementing.

Change-Id: I984341ad6b09fadb8cb37e360e3ea37b029bafd4
Signed-off-by: sub77 <sub77@ymail.com>
---
 stack/btm/btm_inq.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/stack/btm/btm_inq.cc b/stack/btm/btm_inq.cc
index 7f9cb40..11c155f 100644
--- a/stack/btm/btm_inq.cc
+++ b/stack/btm/btm_inq.cc
@@ -855,7 +855,7 @@ tBTM_STATUS BTM_StartInquiry(tBTM_INQ_PARMS* p_inqparms,
       BTM_TRACE_API(
           "BTM:Interleaving: started LE scan, Advancing to next state: %d",
           p_inq->next_state + 1);
-      p_inq->next_state += 1;
+      p_inq->next_state = (btm_inq_state)(p_inq->next_state + 1);
     }
     /* reset next_state if status <> BTM_Started */
     if (status != BTM_CMD_STARTED) p_inq->next_state = BTM_BR_ONE;
@@ -914,7 +914,7 @@ tBTM_STATUS BTM_StartInquiry(tBTM_INQ_PARMS* p_inqparms,
       BTM_TRACE_API(
           "BTM:Interleaving: Started BTM inq, Advancing to next state: %d",
           p_inq->next_state + 1);
-      p_inq->next_state += 1;
+      p_inq->next_state = (btm_inq_state)(p_inq->next_state + 1);
     }
   }
   if (status != BTM_CMD_STARTED) {
-- 
2.9.5

